src/vanilla/shallow.ts

```diff
@@ -1,25 +1,30 @@
-const isIterable = (obj: object): obj is Iterable<unknown> =>
+type IterableLike<T> = Iterable<T> & {
+  entries: () => IterableIterator<[number, T]>
+}
+
+const isIterable = (obj: object): obj is IterableLike<unknown> =>
   Symbol.iterator in obj
 
-const compareMapLike = (
-  iterableA: Iterable<[unknown, unknown]>,
-  iterableB: Iterable<[unknown, unknown]>,
-) => {
-  const mapA = iterableA instanceof Map ? iterableA : new Map(iterableA)
-  const mapB = iterableB instanceof Map ? iterableB : new Map(iterableB)
-  if (mapA.size !== mapB.size) return false
-  for (const [key, value] of mapA) {
-    if (!Object.is(value, mapB.get(key))) {
+const toObject = (value: IterableLike<unknown>) =>
+  Object.fromEntries(value.entries())
```

Comment:
This feels less efficient. I wonder if we can keep the previous logic.

Reply:
@dai-shi I feel that supporting `Map`s, `Set`s, and `Array`s would be complex if we want to keep the previous. I'd say that this implementation is pretty straightforward and we don't need to implement a different logic for iterables.

Reply:
Hmm, but Object.fromEntries feels bad for a large object. You might be right for bundle size though. Let me try and compare the two.

Reply:
I tried #2821, but the implementation is rather tricky (even it might be efficient in some cases). So, if you can fix this #2816, it might look cleaner and better.

Reply:
sure, let me take care of that